src/api/apiHelpers.ts

```diff
@@ -81,23 +81,22 @@ export const safeFetch = async (
   const token = authorizationHeader.replace('Bearer ', '')
   const accessTokenStatus = getTokenStatus(token)
 
-  if (accessTokenStatus === 'unknown') {
-    return createNeedsAuthenticationResponse(url)
-  }
-
-  // If the token is expired, we refresh it before calling the backend
-  if (accessTokenStatus === 'expired') {
+  // If the token is expired or unknown, we refresh it before calling the backend
+  if (accessTokenStatus === 'expired' || accessTokenStatus === 'unknown') {
     try {
       const { result: newAccessToken, error } = await refreshAccessToken(api)
-      if (error === REFRESH_TOKEN_IS_EXPIRED_ERROR) {
-        return RefreshTokenExpiredResponse
-      }
 
-      if (error) {
-        eventMonitoring.captureException(new Error(`safeFetch ${error}`))
-        return createNeedsAuthenticationResponse(url)
+      switch (error) {
+        case REFRESH_TOKEN_IS_EXPIRED_ERROR:
+          return RefreshTokenExpiredResponse
+        case FAILED_TO_GET_REFRESH_TOKEN_ERROR:
+          return createNeedsAuthenticationResponse(url)
+        case UNKNOWN_ERROR_WHILE_REFRESHING_ACCESS_TOKEN:
+          eventMonitoring.captureException(new Error(`safeFetch ${error}`, { cause: error }), {
+            extra: { url },
+          })
+          return createNeedsAuthenticationResponse(url)
```

Comment:
Il faudrait remonter les erreurs qu'on n'a pas défini nous-même aussi (par exemple "Keychain non accessible")
```suggestion
        case undefined:
          break
        case UNKNOWN_ERROR_WHILE_REFRESHING_ACCESS_TOKEN:
        default:
          eventMonitoring.captureException(new Error(`safeFetch ${error}`, { cause: error }), {
            extra: { url },
          })
          return createNeedsAuthenticationResponse(url)
```

Reply:
Est-ce que c'est vraiment utile de faire ça ? Comme on ne veut rien faire dans ce cas là, il va juste pas rentrer dans les autres cas et ça revient au même non ?